p360monitoring.php

```diff
@@ -82,46 +89,32 @@ function p360monitoring_ClientArea(array $params): string
 {
     global $CONFIG;
 
-    $keyId = $params['customfields']['keyId'];
-    $uid = $params['customfields']['activationInfoUid'];
-    $activated = $params['customfields']['activationInfoActivated'];
-    $activationLink = $params['customfields']['activationLink'];
-    $terminated = $params['model']->serviceProperties->get('terminated');
-    $suspended = $params['model']->serviceProperties->get('suspended');
+    $kaApi = p360monitoring_getKaApiClient($params);
+    $keyId = $params['customfields'][CustomFields::KEY_ID];
     $translator = Translator::getInstance($CONFIG);
 
     try {
-        if (!$activated) {
-            $kaApi = p360monitoring_getKaApiClient($params);
-
-            $license = $kaApi->retrieveLicense($keyId);
-            $activationLink = $license->getKeyIdentifiers()->getActivationLink();
-            $activated = $license->getActivationInfo()->isActivated();
-            $uid = $license->getActivationInfo()->getUid();
-            $terminated = $license->isTerminated();
-            $suspended = $license->isSuspended();
-
-            $params['model']->serviceProperties->save([
-                'activationLink' => $activationLink,
-                'activationInfoUid' => $uid,
-                'activationInfoActivated' => $activated,
-                'terminated' => $terminated,
-                'suspended' => $suspended
-            ]);
+        $license = $kaApi->retrieveLicense($keyId);
+        $domain = $params[ProductOptions::DOMAIN];
+
+        if (empty($domain)) {
+            $activationUrl = $license->getKeyIdentifiers()->getActivationLink();
+            $dashboardUrl = 'https://monitoring.platform360.io/dashboard/overview';
+        } else {
+            $activationUrl = 'https://' . $domain . '/license/activate/' . $license->getKeyIdentifiers()->getActivationCode();
+            $dashboardUrl = 'https://' . $domain . '/dashboard/overview';
         }
 
         $returnHtml = '';
 
-        if ($activated) {
+        if ($license->getActivationInfo()->isActivated()) {
             $returnHtml .= '<div class="tab-content"><div class="row"><div class="col-sm-3 text-left">' . $translator->translate('p360monitoring_button_license_activated') . '</div></div></div><br/>';
```

Comment:
if (...)
return html;

$returnHtml = ''
if (!$license->isTerminated() && !$license->isSuspended()) {
...
}
$returnHtml .= '<div ....
retrurn $returnHtml

Reply:
just remove if-else, replace with one if and return

Reply:
Fixed